### PR TITLE
Waitlist for WD

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/0/1202843599939263/f 

### Description
Adds a new waitlist in settings. Similar to the one we had for macOS.

### Steps to test this PR
Follow instructions in https://app.asana.com/0/0/1203975737540623/f

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by cmonfortep** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2831** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2831" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2831?utm_source=stack-comment).

File: macos/macos-api/build.gradle [hunk 1]
```diff
@@ -15,17 +15,21 @@
  */
 
 plugins {
-    id 'java-library'
```
Comment (cmonfortep) [inline macos/macos-api/build.gradle:18]: @marcosholgado I need to navigate to MacOsActivity from my module, see `MacOsNav`. That interface has a dependency with Context and Intent, that's why we need to change this from java-library to android-library.

File: windows/windows-api/src/main/java/com/duckduckgo/windows.api/WindowsSettingsNav.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+package com.duckduckgo.windows.api
+
+import android.content.Context
+import android.content.Intent
+
+/**
+ * Public interface to provide navigation Intents related to Windows App Settings
+ */
+interface WindowsSettingsNav {
+    fun openWindowsSettings(context: Context): Intent
```
Comment (aitorvs) [inline windows/windows-api/src/main/java/com/duckduckgo/windows.api/WindowsSettingsNav.kt:26]: I would rename this to `activityContext` and even probably annotated with `ActivityContext`. That's the only reason you need a context here, if it was the app `context` you would not need it as a parameter

File: macos/macos-api/src/main/java/com/duckduckgo/macos_api/MacOsNav.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+package com.duckduckgo.macos_api
+
+import android.content.Context
+import android.content.Intent
+
+/**
+ * Public interface to provide navigation Intents related to MacOs Settings
+ */
+interface MacOsNav {
+    fun openMacOsSettings(context: Context): Intent
```
Comment (aitorvs) [inline macos/macos-api/src/main/java/com/duckduckgo/macos_api/MacOsNav.kt:26]: Same comment as below, rename to `activityContext ` and annotate with `@ActivityContext`

File: windows/windows-api/src/main/java/com/duckduckgo/windows.api/WindowsSettingsNav.kt (hunk 2 above)
Comment (cmonfortep) [reply windows/windows-api/src/main/java/com/duckduckgo/windows.api/WindowsSettingsNav.kt:26]: Sounds good 👍

File: common-ui/src/main/res/drawable/ic_platform_macos_16.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="16dp"
+    android:height="16dp"
+    android:viewportWidth="16"
+    android:viewportHeight="16">
+  <path
+      android:pathData="M5,4.75C5,4.336 4.664,4 4.25,4C3.836,4 3.5,4.336 3.5,4.75V6.25C3.5,6.664 3.836,7 4.25,7C4.664,7 5,6.664 5,6.25V4.75Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"/>
```
Comment (nalcalag) [inline common-ui/src/main/res/drawable/ic_platform_macos_16.xml:9]: When adding icons, you need to remove alpha and replace color to `?attr/daxColorPrimaryIcon` so they look good in light/dark mode
```suggestion
      android:fillColor="?attr/daxColorPrimaryIcon" />
```

File: common-ui/src/main/res/drawable/ic_platform_macos_16.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  <path
+      android:pathData="M12.5,4.75C12.5,4.336 12.164,4 11.75,4C11.336,4 11,4.336 11,4.75V6.25C11,6.664 11.336,7 11.75,7C12.164,7 12.5,6.664 12.5,6.25V4.75Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"/>
```
Comment (nalcalag) [inline common-ui/src/main/res/drawable/ic_platform_macos_16.xml:13]:
```suggestion
      android:fillColor="?attr/daxColorPrimaryIcon" />
```

File: common-ui/src/main/res/drawable/ic_platform_macos_16.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  <path
+      android:pathData="M3.25,1C1.455,1 0,2.455 0,4.25V11.75C0,13.545 1.455,15 3.25,15H9.24C9.247,15 9.254,15 9.262,15H12.75C14.545,15 16,13.545 16,11.75V4.25C16,2.455 14.545,1 12.75,1H3.25ZM7.7,2.5H3.25C2.283,2.5 1.5,3.283 1.5,4.25V11.75C1.5,12.717 2.283,13.5 3.25,13.5H8.131C8.096,13.386 8.061,13.263 8.025,13.131C7.94,12.81 7.855,12.433 7.78,11.993C5.525,11.902 3.861,10.885 3.236,10.296C2.934,10.012 2.92,9.538 3.204,9.236C3.488,8.934 3.962,8.92 4.264,9.204C4.635,9.553 5.861,10.343 7.59,10.48C7.551,10.03 7.523,9.538 7.509,9H6.75C6.336,9 6,8.664 6,8.25C6,5.701 6.882,3.745 7.7,2.5ZM9.72,13.5C9.65,13.324 9.563,13.076 9.475,12.744C9.412,12.51 9.349,12.233 9.289,11.911C11.075,11.637 12.306,10.796 12.802,10.258C13.082,9.953 13.063,9.479 12.758,9.198C12.453,8.918 11.979,8.937 11.698,9.242C11.432,9.531 10.52,10.198 9.091,10.424C9.035,9.801 9,9.08 9,8.25C9,7.836 8.664,7.5 8.25,7.5H7.53C7.734,4.979 8.921,3.244 9.575,2.5H12.75C13.717,2.5 14.5,3.283 14.5,4.25V11.75C14.5,12.717 13.717,13.5 12.75,13.5H9.72Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"
```
Comment (nalcalag) [inline common-ui/src/main/res/drawable/ic_platform_macos_16.xml:17]:
```suggestion
      android:fillColor="?attr/daxColorPrimaryIcon"
```

File: common-ui/src/main/res/drawable/ic_platform_windows_16.xml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      android:fillAlpha="0.84"/>
+  <path
+      android:pathData="M8.5,1.5H14.5V7.5H8.5V1.5Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"/>
+  <path
+      android:pathData="M14.5,8.5H8.5V14.5H14.5V8.5Z"
+      android:fillColor="#000000"
+      android:fillAlpha="0.84"/>
+</vector>
```
Comment (nalcalag) [inline common-ui/src/main/res/drawable/ic_platform_windows_16.xml:22]: Same as before, you need to replace `fillColor` attribute with `?attr/daxColorPrimaryIcon`  and remove alpha in all `<paths/>`

Comment (cmonfortep) [reply common-ui/src/main/res/drawable/ic_platform_windows_16.xml]: thanks for the check @nalcalag 👍

**Review by marcosholgado — APPROVED** [review]
LGTM!

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/TextExtensions.kt [hunk 8]
```diff
@@ -82,7 +82,7 @@ fun TextView.addClickableLink(
             )
             setSpan(
                 ForegroundColorSpan(
-                    ContextCompat.getColor(context, R.color.cornflowerBlue),
+                    context.getColorFromAttr(R.attr.daxColorAccentBlue),
```
Comment (cmonfortep) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/TextExtensions.kt:84]: updated spannable links colors (reported by sveta, discussed with @malmstein )

**Pull request merged**